integration-tests/deployment/ccip/add_chain.go

```diff
@@ -0,0 +1,170 @@
+					To:    state.Chains[source].OnRamp.Address(),
+					Data:  enableOnRampDest.Data(),
+					Value: big.NewInt(0),
+				},
+				{
+					// Set initial dest prices to unblock testing.
+					To:    state.Chains[source].FeeQuoter.Address(),
+					Data:  initialPrices.Data(),
+					Value: big.NewInt(0),
+				},
+				{
+					To:    state.Chains[source].FeeQuoter.Address(),
+					Data:  enableFeeQuoterDest.Data(),
+					Value: big.NewInt(0),
+				},
+			},
+		})
+		metaDataPerChain[mcms.ChainIdentifier(chain.Selector)] = timelock.MCMSWithTimelockChainMetadata{
+			ChainMetadata: mcms.ChainMetadata{
+				NonceOffset: 0,
+				MCMAddress:  state.Chains[source].Mcm.Address(),
+			},
+			TimelockAddress: state.Chains[source].Timelock.Address(),
+		}
+	}
+
+	// Home chain new don.
+	// - Add new DONs for destination to home chain
+	nodes, err := deployment.NodeInfo(e.NodeIDs, e.Offchain)
+	if err != nil {
+		return nil, err
+	}
+	encodedExtraChainConfig, err := chainconfig.EncodeChainConfig(chainconfig.ChainConfig{
+		GasPriceDeviationPPB:    ccipocr3.NewBigIntFromInt64(1000),
+		DAGasPriceDeviationPPB:  ccipocr3.NewBigIntFromInt64(0),
+		OptimisticConfirmations: 1,
+	})
+	if err != nil {
+		return nil, err
+	}
+	chainConfig := SetupConfigInfo(newChainSel, nodes.NonBootstraps().PeerIDs(),
+		nodes.DefaultF(), encodedExtraChainConfig)
+	addChain, err := state.Chains[homeChainSel].CCIPConfig.ApplyChainConfigUpdates(SimTransactOpts(), nil, []ccip_config.CCIPConfigTypesChainConfigInfo{
+		chainConfig,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	newDONArgs, err := BuildAddDONArgs(e.Logger, state.Chains[newChainSel].OffRamp, e.Chains[newChainSel], nodes.NonBootstraps())
+	if err != nil {
+		return nil, err
+	}
+	addDON, err := state.Chains[homeChainSel].CapabilityRegistry.AddDON(SimTransactOpts(),
+		nodes.NonBootstraps().PeerIDs(), []capabilities_registry.CapabilitiesRegistryCapabilityConfiguration{
+			{
+				CapabilityId: CCIPCapabilityId,
+				Config:       newDONArgs,
+			},
+		}, false, false, nodes.NonBootstraps().DefaultF())
+	if err != nil {
+		return nil, err
+	}
+	homeChain, _ := chainsel.ChainBySelector(homeChainSel)
+	metaDataPerChain[mcms.ChainIdentifier(homeChain.Selector)] = timelock.MCMSWithTimelockChainMetadata{
+		ChainMetadata: mcms.ChainMetadata{
+			NonceOffset: 0,
+			MCMAddress:  state.Chains[homeChainSel].Mcm.Address(),
+		},
+		TimelockAddress: state.Chains[homeChainSel].Timelock.Address(),
+	}
+	batches = append(batches, timelock.BatchChainOperation{
+		ChainIdentifier: mcms.ChainIdentifier(homeChain.Selector),
+		Batch: []mcms.Operation{
+			{
+				// Add the chain first, don needs it to be there.
+				To:    state.Chains[homeChainSel].CCIPConfig.Address(),
+				Data:  addChain.Data(),
+				Value: big.NewInt(0),
+			},
+			{
+				To:    state.Chains[homeChainSel].CapabilityRegistry.Address(),
+				Data:  addDON.Data(),
+				Value: big.NewInt(0),
+			},
+		},
+	})
+	return timelock.NewMCMSWithTimelockProposal(
+		"1",
```

Comment:
Comments on string/bool/int literal fields would be useful

Reply:
yeah lets punt to CCIP-3411